lib/Reader/abisignature.cpp

```diff
@@ -448,8 +450,11 @@ Value *ABICallSignature::emitCall(GenIR &Reader, Value *Target, bool MayThrow,
   const bool IsVarArg = false;
   Type *FunctionTy = FunctionType::get(FuncResultType, ArgumentTypes, IsVarArg);
   Type *FunctionPtrTy = Reader.getUnmanagedPointerType(FunctionTy);
-
-  Target = Builder.CreateIntToPtr(Target, FunctionPtrTy);
+  if (Target->getType()->isPointerTy()) {
+    Target = Builder.CreateBitCast(Target, FunctionPtrTy);
```

Comment:
I'd prefer that this transformation be done up the stack from `::emitCall`, as it deals with IL semantics (and this code doesn't, for the most part). In that case, the assert above would remain `assert(...isIntegerTy(...))`. It might also be good to note the parts of the spec that imply that this is necessary (which I think is just II.14.5 and III.3.20).